ckan/lib/helpers.py

```diff
@@ -1285,9 +1285,14 @@ def linked_user(user: Union[str, model.User],
 @core_helper
 def group_name_to_title(name: str) -> str:
     group = model.Group.by_name(name)
-    if group is not None:
-        return group.display_name
-    return name
+    if group is None:
+        return name
+    group_dict = logic.get_action(u'%s_show' % group.type)(
+                    {}, {u'id': group.id})
+    if group_dict is None:
+        return name
+    return p.toolkit.h.get_translated(group_dict, 'title') \
+        or group_dict.name
```

Comment:
If this is called in a loop like in the facet titles below it's going to be really slow. I don't see what other part of core is using this helper, maybe we should plan to remove it instead?

Reply:
Okay, nothing is going to be using this now. I have added in a deprecation notice for it's removal.